frontend/packages/kubevirt-plugin/src/yaml-templates.ts

```diff
@@ -39,4 +41,76 @@ spec:
           cloudInitNoCloud:
             userDataBase64: SGkuXG4=
 `,
-);
+  )
+  .setIn(
+    [TemplateModel, 'vm-template'],
+    `
+apiVersion: ${TemplateModel.apiGroup}/${TemplateModel.apiVersion}
+kind: ${TemplateModel.kind}
+metadata:
+  name: vm-template-example
+  labels:
+    kubevirt.io/os: fedora27
+    miq.github.io/kubevirt-is-vm-template: 'true'
+    template.kubevirt.io/type: vm
+  annotations:
+    description: VM template example
+    iconClass: icon-fedora
+    tags: 'kubevirt,ocp,template,linux,virtualmachine'
+objects:
+  - apiVersion: kubevirt.io/v1alpha3
+    kind: VirtualMachine
+    metadata:
+      creationTimestamp: null
+      labels:
+        kubevirt-vm: 'vm-\${NAME}'
```

Comment:
I'm probably missing something but where does `NAME` (and other uppercase-labeled variables) come from?

Reply:
It's the template pramters, https://docs.okd.io/latest/dev_guide/templates.html#writing-parameters

default values defined here:
https://github.com/openshift/console/pull/1673/files#diff-a05f509d407deb7b65bb87cd43944154R106

Reply:
@yaacov Thank you for clarification.